readFile is broken for relative paths in parent directories

### Operating system

Windows 11

### Helmfile Version

1.2.3

### Helm Version

4.0.4

### Bug description

Starting with helmfile 1.2.0 (needed for helm v4), any readFile instructions pointing to relative paths via the parent directory are broken.

The files do simply not get copied to the temporary location.

### Example helmfile.yaml


helmfile
```yml
releases:
  name: example-service
  chart: ../services/example-service
  values:
    - ../services/example-service/values/example.yaml.gotmpl
```

filestructure
```
helmfile.d/
  - 01-example-service.yaml
services/
  - example-service/
    - files/default.conf
    - values/example.yaml.gotmpl
```

example.yaml.gotmpl
```
configMaps:
  nginx:
    data:
      default.conf: |-
        {{- readFile "../files/default.conf" | nindent 8 }}
```

### Error message you've seen (if any)

```log
panic: BUG: please file an github issue for this unhandled error: *errors.errorString: failed to render values files "C:\<repo>\services\example-service\values\example.yaml.gotmpl": failed to render [C:\Users\<user>\AppData\Local\helmfile\values\file__\<repo>\services\generic\values\example.yaml.gotmpl], because of template: stringTemplate:42:12: executing "stringTemplate" at <readFile "../files/default.conf">: error calling readFile: open C:\Users\<user>\AppData\Local\helmfile\values\file__\<repo>\services\example-service\files\default.conf: The system cannot find the path specified.

goroutine 46 [running]:
github.com/helmfile/helmfile/cmd.toCLIError(0xc0008f2360?, {0x7ff6f14f14e0?, 0xc0049f4d80?})
        /home/runner/work/helmfile/helmfile/cmd/root.go:38 +0x268
github.com/helmfile/helmfile/cmd.NewTemplateCmd.func1(0xc001070800?, {0x7ff6f0c142f5?, 0x4?, 0x7ff6f0c14405?})
        /home/runner/work/helmfile/helmfile/cmd/template.go:29 +0x117
github.com/spf13/cobra.(*Command).execute(0xc0007e6608, {0xc000727980, 0x4, 0x4})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1015 +0xb02
github.com/spf13/cobra.(*Command).ExecuteC(0xc000f02908)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1148 +0x465
github.com/spf13/cobra.(*Command).Execute(0xc000b1d7c0?)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1071 +0x13
main.main.func1()
        /home/runner/work/helmfile/helmfile/main.go:29 +0x69
created by main.main in goroutine 1
        /home/runner/work/helmfile/helmfile/main.go:21 +0xcf
```

Note: Under `C:\Users\<user>\AppData\Local\helmfile\values\file__\<repo>\services\example-service\` only the "values" folder was copied

### Steps to reproduce

`helmfile lint` with the structure above

### Working Helmfile Version

1.1.9 + helm 3.20.0

### Relevant discussion

_No response_

--- a/pkg/remote/remote.go
+++ b/pkg/remote/remote.go
@@ -102,6 +102,9 @@ type Source struct {
 }
 
 func IsRemote(goGetterSrc string) bool {
+	if filepath.IsAbs(goGetterSrc) {
+		return false
+	}
 	if _, err := Parse(goGetterSrc); err != nil {
 		return false
 	}
@@ -120,6 +123,13 @@ func Parse(goGetterSrc string) (*Source, error) {
 		if len(items) == 2 {
 			return ParseNormal(goGetterSrc)
 		}
+	}
+
+	// Local absolute paths (e.g., C:\path on Windows, /path on Unix)
+	// are not remote URLs. Reject them before go-getter's url.Parse
+	// can misinterpret Windows drive letters as URL schemes.
+	if filepath.IsAbs(goGetterSrc) {
+		return nil, InvalidURLError{err: fmt.Sprintf("parse url: local absolute path is not a remote URL: %s", goGetterSrc)}
 	}
 
 	u, err := url.Parse(goGetterSrc)

--- a/pkg/remote/remote_test.go
+++ b/pkg/remote/remote_test.go
@@ -356,6 +356,54 @@ func TestRemote_S3(t *testing.T) {
 	}
 }
 
+func TestIsRemote(t *testing.T) {
+	testcases := []struct {
+		name     string
+		input    string
+		expected bool
+	}{
+		{
+			name:     "git remote URL",
+			input:    "git::https://github.com/cloudposse/helmfiles.git@releases/kiam.yaml?ref=0.40.0",
+			expected: true,
+		},
+		{
+			name:     "s3 remote URL",
+			input:    "s3://helm-s3-values-example/values.yaml",
+			expected: true,
+		},
+		{
+			name:     "https remote URL",
+			input:    "https://example.com/values.yaml",
+			expected: true,
+		},
+		{
+			name:     "relative path",
+			input:    "relative/path/to/file.yaml",
+			expected: false,
+		},
+		{
+			name:     "parent-relative path",
+			input:    "../services/values.yaml",
+			expected: false,
+		},
+		{
+			name:     "unix absolute path",
+			input:    "/absolute/path/to/file.yaml",
+			expected: false,
+		},
+	}
+
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+			result := IsRemote(tt.input)
+			if result != tt.expected {
+				t.Errorf("IsRemote(%q) = %v, want %v", tt.input, result, tt.expected)
+			}
+		})
+	}
+}
+
 func TestParse(t *testing.T) {
 	testcases := []struct {
 		name   string
@@ -371,6 +419,11 @@ func TestParse(t *testing.T) {
 			name:  "miss scheme",
 			input: "raw/incubator",
 			err:   "parse url: missing scheme - probably this is a local file path? raw/incubator",
+		},
+		{
+			name:  "unix absolute path",
+			input: "/absolute/path/to/file.yaml",
+			err:   "parse url: local absolute path is not a remote URL: /absolute/path/to/file.yaml",
 		},
 		{
 			name:   "git scheme",

--- a/pkg/remote/remote_windows_test.go
+++ b/pkg/remote/remote_windows_test.go
@@ -1,0 +1,69 @@
+//go:build windows
+
+package remote
+
+import (
+	"testing"
+)
+
+func TestIsRemote_Windows(t *testing.T) {
+	testcases := []struct {
+		name     string
+		input    string
+		expected bool
+	}{
+		{
+			name:     "windows drive letter path",
+			input:    `C:\project\services\values.yaml`,
+			expected: false,
+		},
+		{
+			name:     "windows UNC path",
+			input:    `\\server\share\path\values.yaml`,
+			expected: false,
+		},
+	}
+
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+			result := IsRemote(tt.input)
+			if result != tt.expected {
+				t.Errorf("IsRemote(%q) = %v, want %v", tt.input, result, tt.expected)
+			}
+		})
+	}
+}
+
+func TestParse_Windows(t *testing.T) {
+	testcases := []struct {
+		name  string
+		input string
+		err   string
+	}{
+		{
+			name:  "windows drive letter path",
+			input: `C:\project\services\values.yaml`,
+			err:   `parse url: local absolute path is not a remote URL: C:\project\services\values.yaml`,
+		},
+		{
+			name:  "windows UNC path",
+			input: `\\server\share\path\values.yaml`,
+			err:   `parse url: local absolute path is not a remote URL: \\server\share\path\values.yaml`,
+		},
+	}
+
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := Parse(tt.input)
+			if err == nil {
+				t.Fatalf("Parse(%q) expected error, got nil", tt.input)
+			}
+			if _, ok := err.(InvalidURLError); !ok {
+				t.Fatalf("Parse(%q) expected InvalidURLError, got %T: %v", tt.input, err, err)
+			}
+			if err.Error() != tt.err {
+				t.Errorf("Parse(%q) error = %q, want %q", tt.input, err.Error(), tt.err)
+			}
+		})
+	}
+}

